### PR TITLE
docs: add example for cargo-credential

### PR DESCRIPTION
**Pull request by arlosi**
Add additional docs and example for `cargo-credential`.

r? @epage

### CODE REVIEW

File: credential/cargo-credential/Cargo.toml [hunk 1]
```diff
@@ -12,3 +12,7 @@ serde = { workspace = true, features = ["derive"] }
 serde_json.workspace = true
 thiserror.workspace = true
 time.workspace = true
+
+[dev-dependencies]
+snapbox = { workspace = true, features = ["examples"] }
+tempfile.workspace = true
```
Comment (epage) [inline credential/cargo-credential/Cargo.toml:18]: Not seeing this used (and snapbox has some tempfile support)

File: credential/cargo-credential/examples/file-provider.rs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+                // The token for `cargo login` can come from the `login_options` parameter or i
+                // interactively reading from stdin.
+                //
+                // `cargo_credential::read_token` automatically handles this.
+                let token = cargo_credential::read_token(login_options, registry)?;
+                creds.insert(registry.index_url.to_string(), token);
+
+                // `Error::Other` takes a boxed `std::error::Error` type that causes Cargo to
+                // show the error.
+                FileCredential::write(&creds).map_err(cargo_credential::Error::Other)?;
```
Comment (epage) [inline credential/cargo-credential/examples/file-provider.rs:53]: Should we do the same thing when parsing json?

Comment (epage) [reply credential/cargo-credential/examples/file-provider.rs:53]: The risk/complication is that we'd need that error to not get in the way of things like `UrlNotSupported`

File: credential/cargo-credential/tests/examples.rs [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    let hello = r#"{"v":[1]}"#;
+    let login_request = r#"{"v": 1,"registry": {"index-url":"https://github.com/rust-lang/crates.io-index","name":"crates-io"},"kind": "login","token": "s3krit","args": []}"#;
+    let login_response = r#"{"Ok":{"kind":"login"}}"#;
+
+    let get_request = r#"{"v": 1,"registry": {"index-url":"https://github.com/rust-lang/crates.io-index","name":"crates-io"},"kind": "get","operation": "read","args": []}"#;
+    let get_response =
+        r#"{"Ok":{"kind":"get","token":"s3krit","cache":"session","operation_independent":true}}"#;
+
+    Command::new(bin)
+        .current_dir(env!("CARGO_TARGET_TMPDIR"))
```
Comment (epage) [inline credential/cargo-credential/tests/examples.rs:16]: This is the same `CARGO_TARGET_TMPDIR` used by every test in every package in this workspace.  Should we add something to this to make it more unique / descriptive?

Comment (arlosi) [reply credential/cargo-credential/tests/examples.rs]: Added `/cargo-credential-tests`, and automatic cleanup.

File: credential/cargo-credential/examples/file-provider.rs (hunk 2 above)
Comment (arlosi) [reply credential/cargo-credential/examples/file-provider.rs]: The only place we parse JSON is when reading the credential file. I removed the `unwrap_or_default` in favor of better error handling in `read`.

File: credential/cargo-credential/Cargo.toml (hunk 1 above)
Comment (arlosi) [reply credential/cargo-credential/Cargo.toml]: Thanks

**Pull request merged**